tests/admin_views/tests.py

```diff
@@ -67,6 +67,36 @@
 for a staff account. Note that both fields may be case-sensitive."
 
 
+class AdminFieldExtractionMixin(object):
+    """Mixin class for extracting data from iterables of `django.contrib.admin.helpers.AdminForm`."""
+
+    def get_admin_form_fields(self, admin_form):
+        """Extracting Fieldset, FieldLine and AdminField objects from AdminForm class."""
+        fieldsets = list(admin_form)
+
+        field_lines = []
+        for fs in fieldsets:
+            field_lines += list(fs)
+
+        fields = []
+        for field_line in field_lines:
+            fields += list(field_line)
+
+        return fields
+
+    def get_admin_readonly_fields(self, admin_form):
+        """Filtering AdminField list to readonly fields."""
```

Comment:
Return the readonly fields for a given AdminForm.
(following verb style of PEP 0257) -- similar for other docstrings